src/servers/src/proto.rs

```diff
@@ -27,10 +27,13 @@ use crate::prom_store::METRIC_NAME_LABEL_BYTES;
 use crate::repeated_field::{Clear, RepeatedField};
 
 impl Clear for Sample {
-    fn clear(&mut self) {}
+    fn clear(&mut self) {
+        self.timestamp = 0;
+        self.value = 0.0;
```

Comment:
Strange. Setting them to 0 should not affect the final decode result if we correctly implement the repeated vector and the decode phase. Since 0 is also invalid data.

Reply:
Maybe clear means that a struct is reset to its default so we should set them to 0. When decoding the 0 timestamp, `merge()` might consider we already pass a default value into it so it doesn't overwrite that value.